File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.java
class Solution {
    public int[] solution(String[] wallpaper) {
        int[] answer = {0,0,0,0};
        
        int minX = 987654321;
        int minY = 987654321;
        int maxX = -1;
        int maxY = -1;
        
        for(int i = 0; i < wallpaper.length; i++){
            for(int j = 0; j < wallpaper[0].length(); j++){
                if(wallpaper[i].charAt(j) == '#'){
                    minX = Math.min(minX, i);
                    minY = Math.min(minY, j);
                    maxX = Math.max(maxX, i);
                    maxY = Math.max(maxY, j);
                }
            }
        }
        answer[0] = minX;
        answer[1] = minY;
        answer[2] = maxX + 1;
        answer[3] = maxY + 1;
        return answer;
    }
}

File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.js





function solution(wallpaper) {
  var answer = [];
  let minX = Infinity;
  let minY = Infinity;
  let maxX = -1;
  let maxY = -1;

  for(let i = 0; i<wallpaper.length; i++){
    for(let j = 0; j<wallpaper[0].length; j++){
      if(wallpaper[i][j] === '#'){
        if(i < minX) minX = i;
        if(j < minY) minY = j;
        if(i > maxX) maxX = i;
        if(j > maxY ) maxY = j;
      }
    }
  }

  answer = [minX, minY, maxX + 1, maxY + 1];

  return answer;
}